src/plugins/intel_gpu/src/graph/compilation_context.cpp

```diff
@@ -28,8 +30,15 @@ class CompilationContext : public ICompilationContext {
         });
     }
 
-    void push_task(ICompilationContext::Task&& task) override {
-        _queue.push(task);
+    void push_task(size_t key, ICompilationContext::Task&& task) override {
+        std::lock_guard<std::mutex> lock(_mutex);
+        auto iter = _key_map.find(key);
+        if (iter == _key_map.end()) {   // not found
+            auto insert_it = _data_list.insert(_data_list.end(), {key, task});
+            _key_map.insert({key, insert_it});
+        } else {    // found
+            _data_list.splice(_data_list.begin(), _data_list, iter->second);
```

Comment:
why is this needed?

Reply:
It has better performance of first running benchmark_app to use this code when the disk cache is used (cl_cache_dir is set). because the most used kernel is built first.

Reply:
Aligned offline

Reply:
removed it